util/url_test.go

```diff
@@ -10,6 +10,7 @@ func TestIsVstsGitURL(t *testing.T) {
 		url      string
 		expected bool
 	}{
+		{"   https://azure.visualstudio.com/ACR/_git/Build/", false}, // leading spaces cause a parse error
```

Comment:
Can you include the pattern for azure pipelines as well?

Reply:
As a part of https://github.com/Azure/acr-builder/issues/297 we can do that.

Reply:
Updated and put that in this PR as well.